examples/java/src/com/mlang/MValue.java

```diff
@@ -239,14 +245,21 @@ static boolean m_is_defined_false(MValue x) {
     }
   }
 
-  static MValue m_array_index(MValue[] array, int tableStart, int index, int size) {
-    if (index < 0) {
+  static MValue m_array_index(MValue[] array, int tableStart, MValue index, int size) {
+    if (index.isUndefined()) {
+      return mUndefined;
+    } 
+     
+    int indexInteger = (int)index.getValue();
+    
+    if (indexInteger < 0) {
       return zero;
-    } else if (index >= size) {
+    } else if (indexInteger >= size) {
```

Comment:
Shouldn't this be `indexInteger > size`?

Reply:
As discussed, I seem to remember doing it without and having problems, however when running the tests today with a strictly greater than, there are no failures. I've therefore made the suggested change.